src/compile/mark/mark.ts

```diff
@@ -239,8 +239,8 @@ function getStackByFields(model: UnitModel) {
         const fieldDef: FieldDef = channelEncoding;
         const scale = model.scale(channel);
         fields.push(field(fieldDef, {
-          binSuffix: scale && scale.type === ScaleType.ORDINAL ? '_range' : '_start'
-        }));
+          binSuffix: scale && scale.type === ScaleType.ORDINAL ? 'range' : 'start'
+        }) || 'undefined');
```

Comment:
I think the right fix is assigning `field(fieldDef, {...})` to `const _field` first and then only push if it's defined.   This way it will work correctly with your edge case too.  This one seems important to fix now since we're changing code and we shouldn't add new bug (although it was buggy, this is buggy in a different way ... so worth just fix it here.)